protein/src/org/labkey/protein/ProteinController.java

```diff
@@ -271,6 +264,73 @@ public void addNavTrail(NavTree root)
+    {
+        private static final String POTENTIAL_PROTEIN_DATA_REGION = "PotentialProteins";
+
+        private static BiFunction<Container, User, SQLFragment> _containerConditionProvider = (container, user) -> new SQLFragment();
+
+        @Override
+        public String getDataRegionName()
+        {
+            return POTENTIAL_PROTEIN_DATA_REGION;
+        }
+
+        @Override
+        public @Nullable QueryView createView(ViewContext ctx, ProteinSearchForm form, BindException errors)
+        {
+            QueryView proteinsView = null;
+
+            if (form.isShowMatchingProteins())
+            {
+                UserSchema schema = QueryService.get().getUserSchema(ctx.getUser(), ctx.getContainer(), ProteinUserSchema.NAME);
+
+                if (null != schema)
+                {
+                    QuerySettings proteinsSettings = schema.getSettings(ctx, POTENTIAL_PROTEIN_DATA_REGION);
+                    proteinsSettings.setQueryName(ProteinUserSchema.TableType.Sequences.toString());
+                    proteinsView = new QueryView(schema, proteinsSettings, errors)
+                    {
+                        @Override
+                        protected TableInfo createTable()
+                        {
+                            ProteinUserSchema schema = (ProteinUserSchema) getSchema();
+                            return schema.createSequences();
+                        }
+                    };
+                    // Disable R and other reporting until there's an implementation that respects the search criteria
+                    proteinsView.setViewItemFilter(ReportService.EMPTY_ITEM_LIST);
+
+                    proteinsView.setButtonBarPosition(DataRegion.ButtonBarPosition.TOP);
+                    SequencesTableInfo<ProteinUserSchema> sequencesTableInfo = (SequencesTableInfo<ProteinUserSchema>) proteinsView.getTable();
+                    SQLFragment containerCondition = _containerConditionProvider.apply(ctx.getContainer(), ctx.getUser());
+                    ((ProbabilityProteinSearchForm)form).setRestrictCondition(containerCondition);
+                    int[] seqIds = form.getSeqId();
+                    if (seqIds.length <= 500)
+                    {
+                        sequencesTableInfo.addSeqIdFilter(seqIds);
+                    }
+                    else
+                    {
+                        sequencesTableInfo.addProteinNameFilter(form.getIdentifier(), form.isExactMatch() ? MatchCriteria.EXACT : MatchCriteria.PREFIX);
+                        if (form.isRestrictProteins())
+                        {
+                            sequencesTableInfo.addCondition(containerCondition);
+                        }
+                    }
+                    proteinsView.setTitle("Matching Proteins (" + (seqIds.length == 0 ? "None" : seqIds.length) + ")");
+                }
+            }
+
+            return proteinsView;
```

Comment:
I think we can keep the collapse behavior consistent by adding:
```
proteinsView.enableExpandCollapse("ProteinSearchProteinMatches", true);
```
though I haven't tested that.

Reply:
Yeah, that works